src/posix/client.cpp

```diff
@@ -226,7 +228,9 @@ void PrintUsage(const char *aProgramName, FILE *aStream, int aExitCode)
             "    %s [Options] [--] ...\n"
             "Options:\n"
             "    -h  --help                    Display this usage information.\n"
-            "    -I  --interface-name name     Thread network interface name.\n",
+            "    -I  --interface-name name     Thread network interface name.\n"
+            "    -t  --timeout <time>          Time in seconds to actively exit the program. The value may have "
```

Comment:
Rather than adding a timeout to `ot-ctl`, I think we should update our CLI commands to be synchronous by default and only output `Done` when the command itself exits.

We did this a while back for the `ping` command:

https://github.com/openthread/openthread/tree/main/src/cli#ping-async--i-source--m-ipaddr-size-count-interval-hoplimit-timeout

Thoughts?

Reply:
There are commands that are different with `ping`, such as https://github.com/openthread/openthread/tree/main/src/cli#linkmetrics-mgmt-ipaddr-enhanced-ack-register-qmrr, the actual result comes from the peer and the actual result won't output `Done`.  And there is a corner case that if the response message is lost, the cli won't output anything. Adding a parameter `timeout` could handle this case. 

Here is another case, the cli will output Link Metrics data of each received Enh Ack after we registered an Enhanced-ACK Based Probing. If we want to catch the link metrics data of the Enh Ack, adding a parameter `timeout` should be a better option.

Example:
```
> linkmetrics mgmt fe80:0:0:0:c4a2:4d65:14cf:9178 enhanced-ack register qm
Done
> Received Link Metrics data in Enh Ack from neighbor, short address:0x3400 , extended address:c6a24d6514cf9178
 - LQI: 228 (Exponential Moving Average)
 - Margin: 0 (dB) (Exponential Moving Average)
> Received Link Metrics Management Response from: fe80:0:0:0:c4a2:4d65:14cf:9178
Status: Success
> Received Link Metrics data in Enh Ack from neighbor, short address:0x3400 , extended address:c6a24d6514cf9178
 - LQI: 228 (Exponential Moving Average)
 - Margin: 0 (dB) (Exponential Moving Average)
```

I just updated the parameter name from `timeout` to `time`.  I think the parameter name `time` is more appropriate.

Thoughts?

Reply:
`ping` command has both sync (default) and async modes. I think we should update the other CLI commands to follow the same model.

Reply:
I will update commands to add the `async` mode.